core/src/main/java/org/axonframework/common/lock/PessimisticLockFactory.java

```diff
@@ -105,11 +116,92 @@ private DisposableLock lockFor(String identifier) {
         return lock;
     }
 
+    private static Set<Thread> threadsWaitingForMyLocks(Thread owner) {
+        return threadsWaitingForMyLocks(owner, INSTANCES);
+    }
+
+    private static Set<Thread> threadsWaitingForMyLocks(Thread owner, Set<PessimisticLockFactory> locksInUse) {
+        Set<Thread> waitingThreads = new HashSet<>();
+        for (PessimisticLockFactory lock : locksInUse) {
+            lock.locks.values().stream()
+                    .filter(disposableLock -> disposableLock.isHeldBy(owner))
+                    .forEach(disposableLock -> disposableLock.queuedThreads().stream()
+                            .filter(waitingThreads::add)
+                            .forEach(thread -> waitingThreads.addAll(threadsWaitingForMyLocks(thread, locksInUse))));
+        }
+        return waitingThreads;
+    }
+
+    /**
+     * There are 3 values:
+     *
+     * acquireAttempts
+     *  This used to specify the maxium number of attempts to obtain a lock before we back off
+     *  (throwing a {@link LockAcquisitionFailedException}). A value of '-1' means unlimited attempts.
```

Comment:
Suggestion: `...(throwing a {@link LockAcquisitionFailedException} *if it does*).`